### PR TITLE
Add password policy

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | We need to apply a some JS in order to show password policy tips
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Go on register as a customer, type your password and have fun!
| Possible impacts? | Register as a customer

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

File: src/js/components/usePasswordPolicy.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  const element = document.querySelector<HTMLElement>(selector);
+  const elementInput = element?.querySelector<HTMLInputElement>('input');
+  const templateElement = document.createElement('div');
+  const feedbackTemplate = document.querySelector<HTMLTemplateElement>(PasswordPolicyMap.template);
+  let feedbackContainer: HTMLElement | null;
+
+  if (feedbackTemplate && element && elementInput) {
+    templateElement.innerHTML = feedbackTemplate.innerHTML;
+    element.append(templateElement);
+    feedbackContainer = element.querySelector<HTMLElement>(PasswordPolicyMap.container);
```
Comment (mparvazi) [inline src/js/components/usePasswordPolicy.ts:159]:
```suggestion

  if (feedbackTemplate && element && elementInput) {
    templateElement.innerHTML = feedbackTemplate.innerHTML;
    element.append(templateElement);
    const feedbackContainer = element.querySelector<HTMLElement>(PasswordPolicyMap.container);
```

File: src/js/components/usePasswordPolicy.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+        if (passwordRequirementsText && passwordRequirementsScore && passwordRequirementsScore.dataset.translation) {
+          passwordRequirementsText.innerText = sprintf(
+            passwordRequirementsScore.dataset.translation,
+            hints[<string>elementInput.dataset.minscore],
+          );
+        }
+
+        // eslint-disable-next-line max-len, @typescript-eslint/no-non-null-assertion
+        elementInput.addEventListener('keyup', () => watchPassword(elementInput, feedbackContainer!, hints));
```
Comment (mparvazi) [inline src/js/components/usePasswordPolicy.ts:190]:
```suggestion
        elementInput.addEventListener('keyup', () => watchPassword(elementInput, feedbackContainer, hints));
```

File: src/js/components/usePasswordPolicy.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+  if (feedbackTemplate && element && elementInput) {
+    templateElement.innerHTML = feedbackTemplate.innerHTML;
+    element.append(templateElement);
+    feedbackContainer = element.querySelector<HTMLElement>(PasswordPolicyMap.container);
+
+    if (feedbackContainer) {
+      const hintElement = document.querySelector(PasswordPolicyMap.hint);
+
+      if (hintElement) {
```
Comment (mparvazi) [inline src/js/components/usePasswordPolicy.ts:164]:
```suggestion
      if (hintElement && hintElement.innerHTML.trim()) {
```

**Comment by mparvazi** [comment]
Screenshot from 2022-04-06 13-20-31

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Could you remove merge commits ?

**Comment by NeOMakinG** [comment]
I'll leave this as a draft and not fixing things until it's merged on the core side, it will avoid me dealing with rebases and whatever :)

**Comment by atomiix** [comment]
@NeOMakinG looks like there's a conflict 🙃

**Comment by NeOMakinG** [comment]
@Hlavtox considering mparvarsi's review and stuff, can you check and approve/merge if it looks fine for you?

**Comment by NeOMakinG** [comment]
@PrestaShop/prestashop-maintainers can I get a review from someone there? @Hlavtox told me that he was ready to merge it but I would like one more pair of eyes on it

**Review by matthieu-rolland — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Pull request merged**